src/d3a/models/area/__init__.py

```diff
@@ -293,7 +293,7 @@ def next_market(self):
         except IndexError:
             return None
 
-    @cached_property
+    @property
```

Comment:
There are a number of places in the code where @cached_property is used. Why changing it (only) here?

Reply:
Thanks for pointing it out, I just saw that it was used in other places in the same file, and it probably needs to be removed from there too. The problem was that, due to caching, the current_market property did not always return the last past market, but one-before-the-last at some points. I suspect that the same can happen for the next_market for instance, that's why I think we need to removed the cached_property from all the properties of this file.